LiveSched/src/main/java/dev/coms4156/project/livesched/Schedule.java

```diff
@@ -0,0 +1,73 @@
+package dev.coms4156.project.livesched;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Coordinates scheduling given a Task and a List of ResourceTypes.
+ * This class stores the available ResourceTypes and a Task
+ * and modifies their values upon scheduling
+ */
+public class Schedule {
+  private final String scheduleId;
+  private Task task;
+  private List<ResourceType> resourceTypes;
+  private List<ResourceType> candidates;
+
+  /**
+   * Constructs a new Task object with the given parameters.
```

Comment:
We would need extra functions to handle priority of the tasks. I thought the main functionality of this class was to get a set of tasks, their priority, and the availability of the resources to return a 'schedule' - the order the tasks should be done at. What do you guys think?

Reply:
Yeah, I'm not sure how to go about giving a set of tasks to Schedule (maybe after we see MyFileDatabase, we could think of a way), but I guess if Schedule can get them, it can try to find and assign resources to high-priority tasks first. If it does that for all of the tasks, then it will essentially create some kind of 'schedule'.

Reply:
I have a feeling that what we're trying to accomplish here is what one of the RouteController method would be doing -- GET resource and GET task and find resource for task. But I guess since it is a complicated method it might be better to store it in a separate class. I am trying to wrap my head around this. Maybe we are overcomplicating this? I'm not sure.

Reply:
Like what Jungyun said, if we have a method that returns a Schedule Class, that would make this class non-trivial. I can go ahead and try to implement that.

Reply:
Actually I think Schedule Class can just modify availableTime/startTime so it can assign to the appropriate resources

Reply:
Returning could be done by RouteController I think??

Reply:
Maybe we could have requiredResources (just rename resourceList) and assignedResources in Task class, and the Schedule class can update those maps. (It already removes the resources from the list of required resources after assigning resources to task). It can additionally update assignedResources, and the RouteController can get and show which resources were assigned to which task? Does this make sense? Feel free to suggest anything!

Reply:
Oh but still not sure how to go about priority. Maybe we could add that during the second iteration

Reply:
Okay, I used a PriorityQueue that sorts Tasks in the order of its priority and whenever it receives a new Task from the network, it will sort. To assign a resource, it finds all available resource based on availability of the resource and location given a Task.  The Task here can be easily accessed by pq.remove(). List of resources will be sent to client, and client chooses one, received by Schedule. Schedule will get the resourceType and assign the Task to the given resource type. How does this sound?

Reply:
I think it sounds good! But maybe for simplicity, Schedule class can automatically assign resource to task for now, and we can add the part about client choosing one later?

Reply:
Yeah, I think that part is handled by RouteController. I can just add some methods that it can use for the future

Reply:
I agree with Joann's point! Maybe we can implement the choosing resource part in the second iteration and stick with automatic assignment of resource for the first iteration?

Reply:
Yeah sounds good!